File: Dairyncia-dotnet/DTOs/AssignRoleDto.cs
using System.ComponentModel.DataAnnotations;

namespace Dairyncia.DTOs
{
    public class AssignRoleDto
    {
        [Required, EmailAddress]
        public string Email { get; set; }

        [Required]
        public string Role { get; set; } // Admin, Manager, Farmer
    }
}

File: Dairyncia-Spring/src/main/java/com/dairyncia/dto/AssignRoleDto.java
package com.dairyncia.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Email;
import lombok.Data;

@Data
public class AssignRoleDto {
	@NotBlank(message="email is required")
	@Email(message="invalid email format")
	public String Email;
	
	@NotBlank(message="role is required")
	public String Role;
	
	public String ManagerId;
}
